src/modules/navigator/mission.cpp

```diff
@@ -764,9 +767,16 @@ Mission::set_mission_items()
 
 					set_vtol_transition_item(&_mission_item, vtol_vehicle_status_s::VEHICLE_VTOL_STATE_FW);
 
-					/* set position setpoint to target during the transition */
-					// TODO: if has_next_position_item and use_next set next, or if use_heading set generated
-					generate_waypoint_from_heading(&pos_sp_triplet->current, _mission_item.yaw);
+					if (has_next_position_item) {
+						/* got next mission item, update setpoint triplet */
+						mission_item_to_position_setpoint(mission_item_next_position, &pos_sp_triplet->current);
+
+					} else {
+						_mission_item.yaw = _navigator->get_global_position()->yaw;
```

Comment:
This seems like a considerable change. Have you tested it?

Reply:
I can't recall what I tested, so this needs to be validated again now.

Reply:
@RomanBapst What this change does is that it looks ahead in the mission (skipping over camera triggers, etc) and finds the next position set point after the transition. This is untested and I would appreciate if you could review and cross-test. It is definitely "more correct" in theory.